var/spack/repos/builtin/packages/pfunit/package.py

```diff
@@ -14,12 +14,13 @@ class Pfunit(CMakePackage):
 
     homepage = "http://pfunit.sourceforge.net/"
     url      = "https://github.com/Goddard-Fortran-Ecosystem/pFUnit/archive/3.2.9.tar.gz"
+    git      = "https://github.com/Goddard-Fortran-Ecosystem/pFUnit.git"
 
     maintainers = ['citibeth']
 
-    # Currently investigating build fails for v 4.0.0.
-    # See discussion in PR #11642.
-    # version('4.0.0',  sha256='b8b6470f2b1e2b19c164c244c10e803bd69c8da9a6a5a65ba7c479fb8b92a1e1') # noqa: E501
+    version('4.1.9',  tag='v4.1.9', submodules=True)
+    version('4.1.5',  tag='v4.1.5', submodules=True)
+    version('4.0.0',  tag='v4.0.0', submodules=True)
     version('3.3.3',  sha256='9f673b58d20ad23148040a100227b4f876458a9d9aee0f0d84a5f0eef209ced5')
```

Comment:
These older versions will need the `url=` construct because their URLs no longer conform to the current standard.

Reply:
Does url_for_version not take care of that?  I successfully installed 3.3.3 on my system with this recipe, but it's possible that I accidentally used a cached tarball.